core/src/main/scala/kafka/admin/TopicCommand.scala

```diff
@@ -42,329 +47,554 @@ object TopicCommand extends Logging {
+    def hasListOption: Boolean = has(listOpt)
+    def hasDescribeOption: Boolean = has(describeOpt)
+    def hasDeleteOption: Boolean = has(deleteOpt)
+
+    def zkConnect: Option[String] = valueAsOption(zkConnectOpt)
+    def bootstrapServer: Option[String] = valueAsOption(bootstrapServerOpt)
+    def commandConfig: Properties = if (has(commandConfigOpt)) Utils.loadProps(options.valueOf(commandConfigOpt)) else new Properties()
+    def topic: Option[String] = valueAsOption(topicOpt)
+    def partitions: Option[Integer] = valueAsOption(partitionsOpt)
+    def replicationFactor: Option[Integer] = valueAsOption(replicationFactorOpt)
+    def replicaAssignment: Option[Map[Int, List[Int]]] =
+      if (has(replicaAssignmentOpt) && !Option(options.valueOf(replicaAssignmentOpt)).getOrElse("").isEmpty)
+        Some(parseReplicaAssignment(options.valueOf(replicaAssignmentOpt)))
+      else
+        None
+    def rackAwareMode: RackAwareMode = if (has(disableRackAware)) RackAwareMode.Disabled else RackAwareMode.Enforced
+    def reportUnderReplicatedPartitions: Boolean = has(reportUnderReplicatedPartitionsOpt)
+    def reportUnavailablePartitions: Boolean = has(reportUnavailablePartitionsOpt)
+    def reportOverriddenConfigs: Boolean = has(topicsWithOverridesOpt)
+    def ifExists: Boolean = has(ifExistsOpt)
+    def ifNotExists: Boolean = has(ifNotExistsOpt)
+    def excludeInternalTopics: Boolean = has(excludeInternalTopicOpt)
+    def topicConfig: Option[util.List[String]] = valuesAsOption(configOpt)
+    def configsToDelete: Option[util.List[String]] = valuesAsOption(deleteConfigOpt)
 
     def checkArgs() {
+      if (args.length == 0)
+        CommandLineUtils.printUsageAndDie(parser, "Create, delete, describe, or change a topic.")
+
+      CommandLineUtils.printHelpAndExitIfNeeded(this, "This tool helps to create, delete, describe, or change a topic.")
+
+      // should have exactly one action
+      val actions = Seq(createOpt, listOpt, alterOpt, describeOpt, deleteOpt).count(options.has)
+      if (actions != 1)
+        CommandLineUtils.printUsageAndDie(parser, "Command must include exactly one action: --list, --describe, --create, --alter or --delete")
+
       // check required args
-      CommandLineUtils.checkRequiredArgs(parser, options, zkConnectOpt)
+      if (options.has(bootstrapServerOpt) == options.has(zkConnectOpt))
+        throw new IllegalArgumentException("Only one of --bootstrap-server or --zookeeper must be specified")
+
+      if (!options.has(bootstrapServerOpt))
+        CommandLineUtils.checkRequiredArgs(parser, options, zkConnectOpt)
       if (!options.has(listOpt) && !options.has(describeOpt))
         CommandLineUtils.checkRequiredArgs(parser, options, topicOpt)
+      if (has(createOpt) && !has(replicaAssignmentOpt))
+        CommandLineUtils.checkRequiredArgs(parser, options, partitionsOpt, replicationFactorOpt)
+      if (has(alterOpt))
+        CommandLineUtils.checkRequiredArgs(parser, options, partitionsOpt)
```

Comment:
with this we can not alter/delete configs using zk option

Reply:
Thanks, didn't notice this. Will modify this to `if (has(bootstrapServerOpt) && has(alterOpt))`.